### ISSUE TITLE
[java] OverrideBothEqualsAndHashCode ignores records

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#4546]
**Affects PMD Version:** 7.0.0-rc2 and 6.x

**Rule:** OverrideBothEqualsAndHashCode

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#overridebothequalsandhashcode>

**Description:**

**Code Sample demonstrating the issue:**

```java
record Foo(int x) {
   @Override boolean equals(Object o) { return this == o; }
}
```

**Expected outcome:**

PMD should report a violation at line ..., but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
